### PR TITLE
Add prototype selection

### PR DESCRIPTION
**Pull request by DanielSprockett**
This is another big commit, but I will try to make smaller, more incremental changes in the future.

### CODE REVIEW

**Review by tanaes — COMMENTED** [review]
Please pull against the origin / main branch now that the other PRs have been merged! It should simplify this one.

**Comment by DanielSprockett** [comment]
Sorry  -- I'm not 100% sure what you mean. Did this commit help simplify things?

File: resources/scripts/run_prototypeSelection.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
```
Comment (tanaes) [inline resources/scripts/run_prototypeSelection.py:1]: It is also possible to include this code as part of the Snakemake smk file itself, under a `run:` declaration.

File: resources/scripts/run_prototypeSelection.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+labels = open(snakemake.input[1]).read().split()
+df2 = 1 - df # similarity to dissimilarity matrix
+print("The imported distance matrix has " + str(df2.shape[1]) + " elements.")
+pt_min = 2
+pt_max = len(labels) - 1
+
+print("Selecting " + str(pt_min) + " to " + str(pt_max) + " prototypes.")
+
+# convert numpy array to type distance matrix
+from skbio.stats.distance import DistanceMatrix
```
Comment (tanaes) [inline resources/scripts/run_prototypeSelection.py:19]: Best practice is to include all imports together at top of file

**Review by tanaes — APPROVED** [review]
Looks good, thanks for pulling in the updates! 

We might consider just sticking all the code from `run_prototypeSelection.py` into a rule itself, or alternatively sticking it in as a __main__ definition in the `prototypeSelection.py` code, so that we don't need to have two separate files.

**Pull request merged**